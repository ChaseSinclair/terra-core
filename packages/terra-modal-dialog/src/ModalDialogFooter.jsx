import React from 'react';
import PropTypes from 'prop-types';
import './ModalDialogFooter.scss';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};

const ModalDialogFooter = ({ children, ...customProps }) => (
  <div className={'terra-ModalDialog-footer'} {...customProps} >
    {children}
  </div>
  );


ModalDialogFooter.propTypes = propTypes;
ModalDialogFooter.defaultProps = defaultProps;

export default ModalDialogFooter;

