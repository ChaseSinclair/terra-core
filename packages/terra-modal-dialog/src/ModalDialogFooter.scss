@import '~terra-mixins';

.terra-ModalDialog-footer {
  background-color: $terra-grey-5;
  border-top: 1px solid $terra-grey-15;
}

