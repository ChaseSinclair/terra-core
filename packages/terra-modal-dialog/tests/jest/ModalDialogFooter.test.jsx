import React from 'react';
import ModalDialogFooter from '../../src/ModalDialogFooter';

const modalDialogFooter = <ModalDialogFooter>This is content</ModalDialogFooter>;

it('should shallow an open modal', () => {
  const modal = shallow(modalDialogFooter);
  expect(modal).toMatchSnapshot();
});

it('should mount an open modal', () => {
  const modal = mount(modalDialogFooter);
  expect(modal).toMatchSnapshot();
});

it('should render an open modal', () => {
  const modal = render(modalDialogFooter);
  expect(modal).toMatchSnapshot();
});

it('should have class terra-ModalDialog-footer', () => {
  const modal = shallow(modalDialogFooter);
  expect(modal.prop('className')).toEqual('terra-ModalDialog-footer');
});
