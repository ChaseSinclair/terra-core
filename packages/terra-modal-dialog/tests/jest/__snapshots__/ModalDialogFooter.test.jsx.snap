exports[`test should mount an open modal 1`] = `
<ModalDialogFooter>
  <div
    className="terra-ModalDialog-footer">
    This is content
  </div>
</ModalDialogFooter>
`;

exports[`test should render an open modal 1`] = `
<div
  class="terra-ModalDialog-footer">
  This is content
</div>
`;

exports[`test should shallow an open modal 1`] = `
<div
  className="terra-ModalDialog-footer">
  This is content
</div>
`;
